assets/js/modules/analytics/components/common/AccountCreate/index.test.js

```diff
@@ -0,0 +1,185 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	createTestRegistry,
+	fireEvent,
+	provideModules,
+	provideSiteInfo,
+	provideUserAuthentication,
+	provideUserInfo,
+	render,
+} from '../../../../../../../tests/js/test-utils';
+import { mockLocation } from '../../../../../../../tests/js/mock-browser-utils';
+
+import {
+	EDIT_SCOPE,
+	MODULES_ANALYTICS,
+	PROVISIONING_SCOPE,
+} from '../../../datastore/constants';
+import {
+	GTM_SCOPE,
+	MODULES_ANALYTICS_4,
+} from '../../../../analytics-4/datastore/constants';
+import { createCacheKey } from '../../../../../googlesitekit/api';
+import { getKeys, setItem } from '../../../../../googlesitekit/api/cache';
+import AccountCreate from '.';
+
+describe( 'AccountCreate', () => {
+	mockLocation();
+
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		provideModules( registry, [
+			{
+				slug: 'analytics',
+				active: true,
+				connected: true,
+			},
+			{
+				slug: 'analytics-4',
+				active: true,
+				connected: true,
+			},
+		] );
+		provideSiteInfo( registry );
+		provideUserInfo( registry );
+		provideUserAuthentication( registry, {
+			grantedScopes: [ EDIT_SCOPE, PROVISIONING_SCOPE, GTM_SCOPE ],
+		} );
+
+		registry.dispatch( MODULES_ANALYTICS ).receiveGetSettings( {} );
+		registry.dispatch( MODULES_ANALYTICS ).receiveGetAccounts( [] );
+		registry
+			.dispatch( MODULES_ANALYTICS )
+			.finishResolution( 'getAccounts', [] );
+	} );
+
+	it( 'renders correctly in a loading state', () => {
+		registry
+			.dispatch( MODULES_ANALYTICS )
+			.invalidateResolution( 'getAccounts', [] );
+
+		const { container, getByRole } = render( <AccountCreate />, {
+			registry,
+			features: [ 'ga4Reporting' ],
+		} );
+
+		expect( container ).toMatchSnapshot();
+
+		expect( getByRole( 'progressbar' ) ).toBeInTheDocument();
+	} );
+
+	it( 'renders correctly when data has loaded', () => {
+		const { container, getByRole } = render( <AccountCreate />, {
+			registry,
+			features: [ 'ga4Reporting' ],
+		} );
+
+		expect( container ).toMatchSnapshot();
+
+		expect(
+			getByRole( 'button', { name: 'Create Account' } )
+		).toBeInTheDocument();
+	} );
+
+	describe( 'when clicking on Create Account', () => {
+		const accountTicketID = 'abc123';
+
+		let getByRole, waitForRegistry;
+
+		beforeEach( () => {
+			( { getByRole, waitForRegistry } = render( <AccountCreate />, {
+				registry,
+				features: [ 'ga4Reporting' ],
```

Comment:
Bearing in mind the points in the IB about a) optional conditionality depending on whether the feature flag is still present (highlighting the fact it will be removed soon), and b) tests being optional depending on execution time, I've kept these tests to a reasonable minimum, and also have only included `ga4Reporting`-enabled versions, figuring these add the most value from a future-proofing perspective.

Reply:
**Update:** Even trying to keep these tests relatively minimal, I still ran into some complications with what looked like a fairly simple snapshot test, resulting in the additional commits to introduce the `useInstanceId()` mocking utilities. :sweat_smile: